### ISSUE TITLE
Difference between expected and actual output of first example in Retry section of Reference Guide

### ISSUE THREAD

**Issue by PascalSchumacher** [reactor/reactor-core#1374]
According to the reference guide (http://projectreactor.io/docs/core/release/reference/#_retrying) the first retry example:

```java
Flux.interval(Duration.ofMillis(250))
    .map(input -> {
        if (input < 3) return "tick " + input;
        throw new RuntimeException("boom");
    })
    .elapsed() 
    .retry(1)
    .subscribe(System.out::println, System.err::println); 

Thread.sleep(2100);
```

Should produce the following output:

>259,tick 0
>249,tick 1
>251,tick 2
>506,tick 0 
>248,tick 1
>253,tick 2
>java.lang.RuntimeException: boom

But when I run the example with reactor-core version `3.2.0` I get:

>[252,tick 0]
>[252,tick 1]
>[248,tick 2]
>[251,tick 0]
>[251,tick 1]
>[249,tick 2]
>java.lang.RuntimeException: boom

The additional duration mentioned in 

>A new interval started, from tick 0. The additional 250ms duration is coming from the 4th tick, the one that causes the exception and subsequent retry.

does not seem to be present anymore.

**Comment by simonbasle** [comment]
TL;DR: This is because the output was originally generated with a slightly different snippet, where `elapsed()` and `retry(1)` were swapped 😖

Having the `elapsed()` **before** the `retry(1)` means that on the start of the retry attempt, it can only measure the time between `retry` re-subscribing to the source and said source emitting its first tick, which is about 250ms. However, the 4th original tick still introduces a "ghost" delay of 250ms just before that, it is simply not visible in this configuration.

Swapping the 2 around, `elapsed()` now sees events from the perspective of what `retry` outputs, and thus measures a (more correct?) 500ms delay between ticks 3 and 4.

I'll update the snippet in 3.1 and 3.2

**Comment by PascalSchumacher** [comment]
Thanks! 👍
